pkg/sql/sqlbase/structured.go

```diff
@@ -829,6 +829,83 @@ func (desc *TableDescriptor) AllActiveAndInactiveChecks() []*TableDescriptor_Che
 	return checks
 }
 
+// ReplaceColumnVarsInExprWithDummies replaces the occurrences of column names in an expression with
+// dummies containing their type, so that they may be typechecked. It returns
+// this new expression tree alongside a set containing the ColumnID of each
+// column seen in the expression.
+func (desc *TableDescriptor) ReplaceColumnVarsInExprWithDummies(
```

Comment:
i think `ReplaceColumnVarsInExprWithTypeDummy` or something with `Type` in the name. knowing what "dummy" it is is helpful.

Reply:
Also noting that "Dummy" suggests to me that they are unusable, but seem important for type checking phase. also pondering on names here a little bit. `TypePlaceholder` would be my ideal if placeholder wasn't already inferring something else.

Reply:
My intention with "usable" here means that you shouldn't try to `eval` this. I was thinking placeholder, but that seems to overload with SQL placeholders.